### PR TITLE
Adding case action view to case in prepration of action-case merge

### PR DESCRIPTION
**Pull request by creativeappdevamit**
# What and why?

# How to test?

# Trello
https://trello.com/c/jLyGuULE/937-s44-create-new-view-in-case-service-and-make-spring-jpa-alignments

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/CaseAction.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import javax.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.NoArgsConstructor;
+import uk.gov.ons.ctp.response.casesvc.representation.CaseGroupStatus;
+
+@Entity
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
```
Comment (warrenbailey) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/CaseAction.java:13]: I would of expected this to need the @Table annotation to tell it what schema to look in?

```
@Table(name = "case_action", schema = "case")
```

File: src/main/resources/database/changes/release-29/case_action_addition.sql [hunk 2]
```diff
@@ -0,0 +1,63 @@
+-- Adding required indexes for case_action view.
+Create index idx_casegroup_collection_exercise_id
```
Comment (warrenbailey) [inline src/main/resources/database/changes/release-29/case_action_addition.sql:2]: A little picky, but can we have SQL keywords in upper case, as thats the convention elsewhere in this project.

i.e.
```
CREATE INDEX idx_casegroup_collection_exercise_id
```

File: src/main/resources/database/changes/release-29/case_action_addition.sql [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ON casesvc.casegroup USING btree (collection_exercise_id);
+Create index idx_case_active_enrolment
```
Comment (warrenbailey) [inline src/main/resources/database/changes/release-29/case_action_addition.sql:4]: there's a few places in this file, where they could be changed to uppercase

File: src/test/resources/delete-test-data.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Delete from casesvc.caseevent;
```
Comment (warrenbailey) [inline src/test/resources/delete-test-data.sql:1]: same here

**Review by warrenbailey — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/CaseAction.java (hunk 1 above)
Comment (creativeappdevamit) [reply src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/CaseAction.java:13]: Not sure - isn't @Table is the transitional actual table and as it is a view not sure what should it be?

File: src/main/resources/database/changes/release-29/case_action_addition.sql (hunk 2 above)
Comment (creativeappdevamit) [reply src/main/resources/database/changes/release-29/case_action_addition.sql:2]: 👍

File: src/main/resources/database/changes/release-29/case_action_addition.sql (hunk 3 above)
Comment (creativeappdevamit) [reply src/main/resources/database/changes/release-29/case_action_addition.sql:4]: 👍

File: src/test/resources/delete-test-data.sql (hunk 4 above)
Comment (creativeappdevamit) [reply src/test/resources/delete-test-data.sql:1]: 👍

**Review by warrenbailey — APPROVED** [review]

**Pull request merged**